rust/agama-cli/src/server.rs

```diff
@@ -0,0 +1,195 @@
+use clap::Subcommand;
+use reqwest::header::{HeaderMap, HeaderValue, CONTENT_TYPE};
+use std::fs::File;
+use std::io;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+const DEFAULT_JWT_FILE: &str = "/tmp/agama-jwt";
+const DEFAULT_AUTH_URL: &str = "http://localhost:3000/authenticate";
+
+#[derive(Subcommand, Debug)]
+pub enum ServerCommands {
+    /// Login with defined server. Result is JWT stored and used in all subsequent commands
+    Login {
+        #[clap(long, short = 'p')]
+        password: Option<String>,
+        #[clap(long, short = 'f')]
+        file: Option<PathBuf>,
+    },
+    /// Release currently stored JWT
+    Logout,
+}
+
+/// Main entry point called from agama CLI main loop
+pub async fn run(subcommand: ServerCommands) -> anyhow::Result<()> {
+    match subcommand {
+        ServerCommands::Login { password, file } => {
+            // actions to do:
```

Comment:
This comment might be useful as part of the command documentation. Here is not that useful.

Reply:
it was a kind of my todo list, but if it makes sense I can move it to the command doc.